### ISSUE TITLE
post-merge not executed after creating release 

### ISSUE THREAD

**Issue by nilshenrich** [nilshenrich/FileTreeHasher#24]
When creating a release using the gitflow workflow in smartgit, I expected to restore the debugging files for the develop branch after finishing the release.\
Unfortunately the git hook post-merge is not executed when merging main back to develop

**Comment by nilshenrich** [comment]
No, post-merge is called, but somehow the script doesn't enter the if clause

**Comment by nilshenrich** [comment]
> No, post-merge is called, but somehow the script doesn't enter the if clause

It enters the if clause but throws the following error:
```bash
----- POST MERGE -----
Restore example files to app
sed: /flutter_app/lib/templates/contentarea.dart kann nicht gelesen werden: Datei oder Verzeichnis nicht gefunden
sed: /flutter_app/lib/templates/contentarea.dart kann nicht gelesen werden: Datei oder Verzeichnis nicht gefunden
Commit restored example files
fatal: Diese Operation muss in einem Arbeitsverzeichnis ausgeführt werden.
fatal: Diese Operation muss in einem Arbeitsverzeichnis ausgeführt werden.
--- POST MERGE END ---
```

**Comment by nilshenrich** [comment]
> > No, post-merge is called, but somehow the script doesn't enter the if clause
> 
> It enters the if clause but throws the following error:
> 
> ```shell
> ----- POST MERGE -----
> Restore example files to app
> sed: /flutter_app/lib/templates/contentarea.dart kann nicht gelesen werden: Datei oder Verzeichnis nicht gefunden
> sed: /flutter_app/lib/templates/contentarea.dart kann nicht gelesen werden: Datei oder Verzeichnis nicht gefunden
> Commit restored example files
> fatal: Diese Operation muss in einem Arbeitsverzeichnis ausgeführt werden.
> fatal: Diese Operation muss in einem Arbeitsverzeichnis ausgeführt werden.
> --- POST MERGE END ---
> ```

`git rev-parse --show-toplevel` throws error and returns nothing

**Comment by nilshenrich** [comment]
With d45ad46ef3fe1ef79193af3a7d395debee8748d8 the execution is correct, but somehow bash doesn't enter the if clause

**Comment by nilshenrich** [comment]
When triggered by git, the current and merged branch are both "develop". But when triggering manually on develop merged main, the merged branch is main as expected

**Comment by nilshenrich** [comment]
git-flow "Finish Release":
```console
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh fetch --progress --prune --force --recurse-submodules=no origin refs/heads/main:refs/remotes/origin/main
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh fetch --progress --prune --force --recurse-submodules=no origin refs/heads/develop:refs/remotes/origin/develop
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh checkout --ignore-other-worktrees main
	Switched to branch 'main'
	Your branch is up to date with 'origin/main'.
	----- POST CHECKOUT -----
	--- POST CHECKOUT END ---
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh merge --no-ff -m "Testing release: Fix view issues on pages" release/v0.0.0+6
	Merge made by the 'ort' strategy.
	 .github/workflows/build.yml                     |  4 ++--
	 flutter_app/lib/definies/datatypes.dart         |  2 +-
	 flutter_app/lib/definies/defaults.dart          |  2 +-
	 flutter_app/lib/definies/hashalgorithms.dart    |  2 +-
	 flutter_app/lib/definies/info.dart              |  4 ++--
	 flutter_app/lib/definies/styles.dart            |  2 +-
	 flutter_app/lib/functions/general.dart          |  2 +-
	 flutter_app/lib/functions/hashfile.dart         |  2 +-
	 flutter_app/lib/main.dart                       |  2 +-
	 flutter_app/lib/templates/contentarea.dart      |  2 +-
	 flutter_app/lib/templates/filetree.dart         |  2 +-
	 flutter_app/lib/templates/hashselector.dart     |  2 +-
	 flutter_app/lib/templates/headercontroller.dart |  2 +-
	 flutter_app/pubspec.yaml                        |  2 +-
	 website/AddBundleDownload.sh                    | 15 +++++++++++----
	 15 files changed, 27 insertions(+), 20 deletions(-)
	========== DEBUG HOOK (post-merge) ==========
	current branch = main
	merged branch  = main
	----- POST MERGE -----
	--- POST MERGE END ---
	~/Dokumente/GitRepos/FileTreeHasher> git tag -f -F /tmp/smartgit10926915129111278362/commit9899805197971248362.tmp v0.0.0+6 refs/heads/main
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh checkout --ignore-other-worktrees develop
	Switched to branch 'develop'
	Your branch is up to date with 'origin/develop'.
	----- POST CHECKOUT -----
	--- POST CHECKOUT END ---
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh merge --no-ff -m "Testing release: Fix view issues on pages" v0.0.0+6
	Merge made by the 'ort' strategy.
	 flutter_app/lib/definies/datatypes.dart         | 2 +-
	 flutter_app/lib/definies/defaults.dart          | 2 +-
	 flutter_app/lib/definies/hashalgorithms.dart    | 2 +-
	 flutter_app/lib/definies/info.dart              | 4 ++--
	 flutter_app/lib/definies/styles.dart            | 2 +-
	 flutter_app/lib/functions/general.dart          | 2 +-
	 flutter_app/lib/functions/hashfile.dart         | 2 +-
	 flutter_app/lib/main.dart                       | 2 +-
	 flutter_app/lib/templates/contentarea.dart      | 2 +-
	 flutter_app/lib/templates/filetree.dart         | 2 +-
	 flutter_app/lib/templates/hashselector.dart     | 2 +-
	 flutter_app/lib/templates/headercontroller.dart | 2 +-
	 flutter_app/pubspec.yaml                        | 2 +-
	 13 files changed, 14 insertions(+), 14 deletions(-)
	========== DEBUG HOOK (post-merge) ==========
	current branch = develop
	merged branch  = develop
	----- POST MERGE -----
	--- POST MERGE END ---
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh push --porcelain --progress origin refs/heads/develop:refs/heads/develop
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Delta compression using up to 16 threads
	Total 22 (delta 17), reused 0 (delta 0), pack-reused 0
	To https://github.com/nilshenrich/FileTreeHasher.git
	  refs/heads/develop:refs/heads/develop 1a59bac..725ccab
	Done
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Delta compression using up to 16 threads
	Total 22 (delta 17), reused 0 (delta 0), pack-reused 0
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh push --porcelain --progress origin refs/heads/main:refs/heads/main
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
	To https://github.com/nilshenrich/FileTreeHasher.git
	  refs/heads/main:refs/heads/main a3f5824..df9c492
	Done
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
	~/Dokumente/GitRepos/FileTreeHasher> git -c credential.helper=/app/extra/smartgit/lib/credentials.sh push --porcelain --progress origin refs/tags/v0.0.0+6:refs/tags/v0.0.0+6
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
	To https://github.com/nilshenrich/FileTreeHasher.git
	* refs/tags/v0.0.0+6:refs/tags/v0.0.0+6 [new tag]
	Done
	/app/extra/smartgit/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/smartgit/git/libexec/git-core/git-remote-https)
	Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
	~/Dokumente/GitRepos/FileTreeHasher> git branch -D release/v0.0.0+6
	Deleted branch release/v0.0.0+6 (was a8c83e1).
```

**Comment by nilshenrich** [comment]
Done with 71e928e8f05644798e9b6216bba28082f3be21a0
